### ISSUE TITLE
[Source Code Confusion]What's the principle of the method 'fill' implementation?

### ISSUE THREAD

**Issue by AaronWharton** [cupcake/sigil#6]
https://github.com/cupcake/sigil/blob/6bf9722f2ae82f58a0c7d48daeb40ec844b78eae/gen/sigil.go#L49
I'm honoured to read your source code, but when I tried understanding the method `fill` I got something puzzled. How does the judgment `data[cell/8]>>uint(8-((cell%8)+1))&1 == 0` can uniquely determin whether the cell will be colored or not? If the number 8 is replaceable?

**Comment by azaghal** [comment]
I haven't looked into whether the Sigil implementation had changed in the meantime (I don't think so), but you might find it useful to have a look at some docs I wrote for the Python port [Pydenticon](https://pydenticon.readthedocs.io/en/0.3/algorithm.html) if you are curious about algorithm itself.

I've tried to describe (for my own sanity) in a non-code way how it works. Provided that is what you are after and not details/info about the actual Go implementation in Sigil.
